chapter04/ex_4_1.pl

```diff
@@ -0,0 +1,20 @@
+#!perl
+use warnings;
+#use strict;
+
+sub total {
+  my $sum;  
+  foreach (@_) {
+    $sum += $_;
+  }
+  $sum;
+}
+
+my @fred = qw{ 1 3 5 7 9 };
+my $fred_total = total(@fred);
+print "The total of \@fred is $fred_total.\n";
+print "Enter some numbers on separate lines: ";
+my $user_total = total(<STDIN>);
+print "The total of those numbers is $user_total.\n";
+
+printf ("Checking string %d", &total("sxg","abc",2));

```

Comment:
<code> printf ("Checking string %d", &total("sxg","abc",2)) </code> Not sure if this was part of error checking, fails nonetheless.